add_cart_s-information.php

```diff
@@ -32,14 +35,62 @@
 
         // Fetch the result
         $result = $stmt->fetch(PDO::FETCH_ASSOC);
-
+        /*
         if ($result) {
             // Return the result as JSON
+            $username = $_SESSION['username']; 
+            $type = $_SESSION['type']; 
+
+            $sqlUser = "SELECT b_id FROM buyer WHERE username = : username"; 
+            $stmt = $conn->prepare($sqlUser);
+            $stmt->bindParam(":username", $username, PDO::PARAM_STR); 
+            $stmt->execute(); 
+            $userResult = $stmt->fetch(PDO::FETCH_ASSOC); 
+            if (!$userResult) {
+                echo json_encode(["error"=> "Cannot find user. "]);
+                exit(); 
+            }
+            $userId = $userResult["b_id"]; 
+            $price = $result['price']; 
+            if ($type === 'vip') {
+                $price *= 0.9; // vip user discount
+            }
+            $orderid = $_SESSION['order_id']; 
+
+            $sqlCheck = "SELECT num_dishes FROM order_detail WHERE order_id = :order_id AND dish_id = :dish_id"; 
+            $stmtCheck = $conn->prepare($sqlCheck);
+            $stmtCheck->bindParam(":order_id", $orderid, PDO::PARAM_STR); 
+            $stmtCheck->bindParam("dish_id", $dishId, PDO::PARAM_STR); 
+            $stmtCheck->execute();
+            $orderDetail = $stmtCheck->fetch(PDO::FETCH_ASSOC);
+            if ($orderDetail) {
+                $numDishes = $orderDetail["num_dishes"]+1;
+                $sqlUpdate = "UPDATE order_detail SET num_dishes = :num_dishes, price = :price WHERE order_id = :order_id AND dish_id = :dish_id";
+                $stmtUpdate = $conn->prepare($sqlUpdate);
+                $stmtUpdate->bindParam("num_dishes", $numDishes, PDO::PARAM_INT); 
+                $stmtUpdate->bindParam("price", $price, PDO::PARAM_INT); 
+                $stmtUpdate->bindParam("order_id", $orderid, PDO::PARAM_STR); 
+                $stmtUpdate->bindParam("dish_id", $price, PDO::PARAM_STR); 
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Incorrect parameter binding in SQL update</b></div>
<div id="fix">

The `bindParam` for `dish_id` parameter appears to be binding `$price` instead of `$dishId`. This could lead to incorrect data being updated in the database.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
                $stmtUpdate->bindParam("dish_id", $dishId, PDO::PARAM_STR);
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Lux-Jason/rms_new/pull/9#issuecomment-2551093127>#46edb8</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged